src/gui/loginrequireddialog/oauthloginrequiredwidget.cpp

```diff
@@ -67,4 +85,25 @@ QList<QPair<QAbstractButton *, QDialogButtonBox::ButtonRole>> OAuthLoginRequired
     };
 }
 
+void OAuthLoginRequiredWidget::showErrorLabelAndRetryButton()
+{
+    _ui->errorFrame->show();
+
+    // apparently, we need to run this in the main event loop since the buttons are owned by the layout within the login required dialog
+    QTimer::singleShot(0, [this]() {
```

Comment:
@erikjv  can you have a look with @fmoc why this hack is needed?
I guess there is a deeper issue here.

Reply:
What is the order in which this method is called, and when `buttons()` is called? I mean, in the `buttons()` method, the ownership of the widget is transferred to the caller. So I'm wondering if this method is called before the layout adds the buttons to itself, and somehow invalidate its state.